rs/state_manager/src/tip.rs

```diff
@@ -1003,8 +1003,6 @@ fn serialize_canister_to_tip(
             metrics,
         )?;
 
-    // Priority credit must be zero at this point
-    assert_eq!(canister_state.scheduler_state.priority_credit.get(), 0);
```

Comment:
This appears to have (partly) been a proxy for `canister_state.scheduler_state.long_execution_mode == LongExecutionMode::default()`. Could we add that check here? Or does it not always hold?